### DISCUSSION TITLE
Edit .xslx in place preserving Excel formatting?

### DISCUSSION THREAD

**Discussion by hoclun-rigsep** [saulpw/visidata#1705] [Q&A]
I don't suppose there is a way to do this so I can better accommodate the normies with whom I share an office?
We have excel files that ought to be database tables and that have been formatted with fonts, font sizes, and so on, for purpose of printing them out.

**Comment by frosencrantz** [comment]
@hoclun-rigsep what happens now when you edit an xslx file and save it?  Does all the formatting get removed or is not included?

**Comment by hoclun-rigsep** [comment]
I mean, my present understanding is that a new xlsx would be created and used to overwrite the existing one? Such that the result would be the same regardless whether the data had come from an xlsx or some other source.

**Comment by reagle** [comment]
I'm not authority -- I find the file saving process confusing -- but the [supported formats page](https://www.visidata.org/docs/formats/) doesn't even list a visidata_saver for xlsx. So I'm not sure what is going on, but even though it can save, I think it is very limited.

**Comment by saulpw** [comment]
There is an xlsx saver, but it is very limited, and is a complete rewrite of the raw data, instead of an update of the existing data.  We could make it updateable, as SQL databases and some other special sheets are, by making it a deferred-modification sheet and supporting `commit-sheet` (`z Ctrl+S`).  This might be reasonable for the basic case, as it seems that the `openpyxl` module we use would support it, although it would likely have a significant impact on performance for large sheets (we use `read_only` and `data_only=True` currently, to maximize loading speed).  But primarily I think there would be a neverending stream of problems with it, as people tried to use it for their own special case and then discovered that it lost this data or that macro or who-knows-what, and that seems like a nightmare to support. 
So I myself am not interested in putting those dozens or maybe hundreds of hours of work into it, especially because **VisiData is not a spreadsheet**--even if it looks similar and can read Excel files.
